examples/cmd/main.go

```diff
@@ -143,39 +209,80 @@ func (c *DownloaderCmd) ExecuteBody(logger log.Logger, fn bodyFunc) error {
 		if err = database.Migrate(c.DB); err != nil && err != migrate.ErrNoChange {
 			return err
 		}
+	}
 
-		downloader, err = github.NewDownloader(client, db)
+	downloadersPool, err := c.buildDownloadersPool(logger, db)
+	if err != nil {
+		return err
 	}
 
-	rate0, err := downloader.RateRemaining(context.TODO())
+	err = downloadersPool.Begin()
 	if err != nil {
 		return err
 	}
-	t0 := time.Now()
 
-	err = fn(client, downloader)
+	err = fn(logger, downloadersPool)
 	if err != nil {
 		return err
 	}
 
-	err = downloader.SetCurrent(c.Version)
+	err = c.commit(downloadersPool)
 	if err != nil {
 		return err
 	}
 
-	if c.Cleanup {
-		return downloader.Cleanup(c.Version)
+	stats, err := downloadersPool.End()
+
+	logger = logger.With(log.Fields{"total-elapsed": stats.Elapsed})
+	logger.Infof("all metadata fetched")
+	for id, ru := range stats.RatesUsage {
+		logger.With(log.Fields{
+			"rate-limit-used":  ru.Used,
+			"rate-usage-speed": fmt.Sprintf("%f/min", ru.Speed),
+		}).Infof("token %d usage", id)
 	}
 
-	elapsed := time.Since(t0)
+	return nil
+}
 
-	rate1, err := downloader.RateRemaining(context.TODO())
-	if err != nil {
-		return err
+func (c *DownloaderCmd) commit(dp *DownloadersPool) error {
+	return dp.WithDownloader(func(d *DownloaderPoolItem) error {
+		var err error
+		err = d.SetCurrent(c.Version)
+		if err != nil {
+			return err
+		}
+
+		if c.Cleanup {
+			return d.Cleanup(c.Version)
+		}
+
+		return nil
+	})
+}
+
+func (c *DownloaderCmd) buildDownloadersPool(logger log.Logger, db *sql.DB) (*DownloadersPool, error) {
+	var clients []*http.Client
+	tokens := strings.Split(c.Tokens, ",")
+	for _, t := range tokens {
+		client := oauth2.NewClient(context.TODO(), oauth2.StaticTokenSource(
+			&oauth2.Token{AccessToken: t},
+		))
+		if c.LogHTTP {
+			setLogTransport(client, logger)
+		}
+
+		clients = append(clients, client)
 	}
-	rateUsed := rate0 - rate1
 
-	logger.With(log.Fields{"rate-limit-used": rateUsed, "total-elapsed": elapsed}).Infof("All metadata fetched")
+	var builder downloaderBuilder
+	if db == nil {
```

Comment:
same comment as before. It's not clear to my why downloader pool should know about db.